src/components/course/EnrollButton.jsx

```diff
@@ -131,176 +71,124 @@ export default function EnrollButton() {
     () => hasCourseStarted(start),
     [start],
   );
-
-  const isUserEnrolled = useMemo(
-    () => isUserEnrolledInCourse({ userEnrollments, key }),
+  const userEnrollment = useMemo(
+    () => findUserEnrollmentForCourse({ userEnrollments, key }),
     [userEnrollments, key],
   );
+  const enrollmentUrl = useCourseEnrollmentUrl({
```

Comment:
Talk to me about this choice. I'm not convinced that this needs to be a hook unless we plan on using it in multiple places. (fwiw, I don't think it's worth undoing now that it's done, I'm just interested in understanding your thought process.)

Reply:
I decided to move it to a hook since the previous implementation was all in a single `useMemo`, resulting in a long dependency array which becomes difficult to reason about. By abstracting it out into a custom hook, we can avoid the lengthy dependency array and imo results in a cleaner implementation when using it.

To me, even if a custom hook isn't re-used in multiple places, if it can remove some of logic in the consuming component resulting in a more readable implementation, I don't see why that logic shouldn't be abstracted into a custom hook.

Reply:
Yeah, that makes sense to me. You can also test the hook separately, which I definitely like. Thanks for your thoughts!